glide-core/src/client/value_conversion.rs

```diff
@@ -226,6 +242,23 @@ fn convert_array_to_map(
     Ok(Value::Map(map))
 }
 
+fn convert_flat_array_to_key_value_pairs(array: &[Value]) -> RedisResult<Value> {
+    if array.len() % 2 != 0 {
```

Comment:
you can move len check to line 184

Reply:
Wouldn't it make more sense to leave it here? If we call this function from other places in the future, we will also need to ensure the array has an even-numbered length. Having the check here instead of outside of this function means we don't need to perform this check multiple times if the function is called in multiple places.